GoogleCloudPlatform/nodejs-docs-samples
# migrate code from googleapis/nodejs-containeranalysis

- added directory structure
- npm init
- moved over samples
- passing tests for create, get, delete note and create occurrence
- passing get occurrence test
- passing delete occurrence
- passing test for occurrences for note, flaky
- passing occurrences for image
- samples
- passing get Discovery Info
- passing find vulnerabilities for image
- passing high severity vulnerability test
- pass all vulnerabilities for image
- failing pubsub
- passing pubsub tests
- passing polling test
- updated imageUrl to https://gcr.io/my-project/my-image:123 for more realistic use case
- removed escape slashes from filter strings, not needed
- updated description of occurrenceId
- added comment to explain second filter statement, needed for testing
- explicity state length greater than zero
- removed system-test, moved tests into test directory
- removed async functions from tests without await
- removed unused exec
- removed explicit project id from pub sub
- ran fix
- replaced promise/timeout with p-retry
- changed all samples to be wrapped with an outer main function, added params as comments for user
- added after step to delete occurrences and notes
- updated polling sample and test
- ran fix, dangling errs for symlink and polling
- updated comment on occurrenceId
- fixed typo in discovery test
- grabs projectId from client, update projectId and vars the refer to it in before hooks, added uuid to test resourceUrl
- removed commented code from polling sample
- move end region tag to inside main function
- removed symlinks from test script, removed commented test from polling block
- deleted unused import from pub sub sample
- changed grafeas to container analysis v1beta1 client
- fixed typo in client, removed comment
- added retry logic to get occurrence test
- changed assert match to assert include
- added create topic logic to fix ci test error
- removed regexp from assert include
- changed tests to include arrow functions
- moved containeranalysis and pubsub into prod deps
- added retry logic for all occs of note, check if topic exists and fixed typo on delete occ test
- added sample-metadata
- fixed occurrence typo
- added params to sample-metadata usage
- added p-retry to prod deps, updated poll sample
- working poll discovery occurrence finished sample
- updated default value of retries and timeoutSeconds to be more readable
- refactor: changes formatting of various statements
- chore: release 1.0.0 (#33)
- fix(deps): update dependency @google-cloud/pubsub to ^0.29.0 (#36)
- chore: release 1.0.1 (#37)
- chore: release 1.0.2 (#40)
- chore: release 1.1.0 (#44)
- feat: adds getGrafeasClient() method on ContainerAnalysisClient instance (#46)
- chore: release 1.2.0 (#47)
- chore: update samples to reflect v1 Grafeas API (#48)
- chore: release 1.2.1 (#50)
- chore: release 1.2.2 (#52)
- docs(samples): Update Topic ID in PubSub sample (#55)
- chore: release 1.3.0 (#58)
- fix(deps): update dependency @google-cloud/pubsub to ^0.30.0 (#59)
- chore: release 1.3.1 (#61)
- chore: release 1.3.2 (#68)
- fix(deps): update dependency @google-cloud/pubsub to ^0.31.0 (#71)
- chore: release 1.3.3 (#72)
- fix(deps): update dependency @google-cloud/pubsub to ^0.32.0 (#81)
- chore: release 1.4.0 (#78)
- fix(deps): update dependency @google-cloud/pubsub to v1 (#84)
- chore: release 1.5.1 (#97)
- chore: add license headers for samples (#106)
- chore: release 1.6.0 (#110)
- chore: release 1.6.1 (#113)
- refactor: use explicit mocha imports (#119)
- chore: release 1.7.0 (#115)
- chore(deps): update dependency mocha to v7 (#121)
- chore: release 1.7.1 (#131)
- feat: move API to typescript code generation (#135)
- chore: release 1.8.0 (#137)
- chore(deps): update dependency uuid to v7
- chore: release 1.9.0 (#144)
- chore: release 1.10.0 (#153)
- feat!: drop node8 support, support for async iterators (#167)
- test: there's a race condition emptying queue (#175)
- test: retry pubsub test, as it relies on many moving parts (#186)
- test: retry queue deletion (#191)
- build: adopt CI changes and generator formatting (#197)
- test: make tests idempotent (#201)
- fix(samples): should use effective_severity field (#200)
- chore: release 2.0.0 (#170)
- build: modified export in protos.js
- chore(deps): update dependency uuid to v8 (#206)
- fix(deps): update dependency @google-cloud/pubsub to v2 (#215)
- chore(deps): update dependency mocha to v8 (#222)
- test: attempt retry on note delete (#224)
- chore: release 2.1.0 (#210)
- chore: release 2.1.1 (#232)
- docs(samples): update sample to illustrate AR style urls
- chore: release 3.0.0 (#264)
- chore: release 3.0.1 (#268)
- chore: release 3.1.0 (#278)
- build: update gapic-generator-typescript to v1.2.10. (#284)
- chore: release 3.1.1 (#294)
- chore: release 3.1.2 (#299)
- chore: new owl bot post processor docker image (#300)
- chore: release 3.1.3 (#303)
- chore: release 3.1.4 (#309)
- chore: release 3.1.5 (#319)
- chore: release 3.1.6 (#325)
- chore: release 3.1.7 (#327)
- chore: release 3.1.8 (#337)
- chore: release 3.2.0 (#340)
- chore: release 3.3.0 (#353)
- chore(main): release 3.4.0 (#389)
- fix: Flaky pubsub test (#394)
- build!: update library to use Node 12 (#408)
- chore(main): release 4.0.0 (#407)
- fix(deps): update dependency @google-cloud/pubsub to v3 (#410)
- chore(main): release 4.1.0 (#416)
- fix(issue-417): adding additional asserts to track flaky error (#421)
- chore(main): release 4.2.0 (#422)
- chore(deps): update dependency uuid to v9 (#435)
- test(samples): disable samples tests for this repository (#440)
- chore(main): release 4.3.0 (#429)
- chore(main): release 4.4.0 (#444)
